src/Domain/Neighbors.cpp

```diff
@@ -0,0 +1,47 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Neighbors.hpp"
+
+#include <algorithm>
+#include <charm.h>
+#include <ostream>
+#include <pup.h>
+
+#include "Domain/ElementId.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+template <size_t VolumeDim>
+Neighbors<VolumeDim>::Neighbors(std::unordered_set<ElementId<VolumeDim>> ids,
+                                Orientation<VolumeDim> orientation)
+    : ids_(std::move(ids)), orientation_(std::move(orientation)) {}
+
+template <size_t VolumeDim>
+void Neighbors<VolumeDim>::add_ids(
+    const std::unordered_set<ElementId<VolumeDim>>& additional_Ids) {
+  for (const auto& id : additional_Ids) {
+    ids_.insert(id);
+  }
+}
+
+template <size_t VolumeDim>
+std::ostream& operator<<(std::ostream& os, const Neighbors<VolumeDim>& n) {
+  os << "Ids = " << n.ids() << "; orientation = " << n.orientation();
+  return os;
+}
+
```

Comment:
add implementation of operator== (use the accessor functions as in operator<< so you don't need to declare it as a friend)